### PR TITLE
Feature/#6369 display message for responsibility free users

### LINKED ISSUE #6369: Give user a message when they have no responsibilities
In the test environment, users with the Initial Authorizer (e.g., mibarra-ia) role are able to login to ECMPS 2.0 with the CBSI Stage bypass enabled (screenshot 1).  When no facilities are returned for ECMPS 2.0 users, these users will receive the following message:

   You have no facilities for which you are responsible. Please login to CBS to review and add facilities for which you are responsible.

**Screenshot 1**
Image

### PR DESCRIPTION
**Pull request by EvanSun220**
## Pull Request

```

- Added related logic in component AboutHome to display warning message on the ECMPS home page, to remind user with IA role and no responsibility assigned about login to CBS to review and add facilities.

```

### CODE REVIEW

File: src/components/AboutHome/AboutHome.js [hunk 1]
```diff
@@ -29,6 +30,8 @@ const AboutHome = ({ user, setCurrentLink }) => {
   const [monitorPlanContent, setMonitorPlanContent] = useState();
   const [qaCertificationContent, setQACertificationContent] = useState();
 
+  const [warningMessage, setWarningMessage] = useState(null);
```
Comment (yonatan-dp) [inline src/components/AboutHome/AboutHome.js]: This state definition is not being used anywhere in a meaningful way (we can remove this). See comment below.

File: src/components/AboutHome/AboutHome.js [hunk 2]
```diff
@@ -49,6 +52,21 @@ const AboutHome = ({ user, setCurrentLink }) => {
...
+    //get the current user
+    const cdxUser = JSON.parse(localStorage.getItem("ecmps_user"));
+
+    //check if the user has initial authorizer role with no responsibilities
+    if (cdxUser?.roles?.includes(config.app.initialAuthorizerRole) &&  JSON.stringify(cdxUser?.facilities) === '{}')
+    {
+      //display the warning message on the home page
+      const warningMessage = 'You have no facilities for which you are responsible. Please login to CBS to review and add facilities for which you are responsible.';
+      displayAppWarning(warningMessage);
+      setWarningMessage(warningMessage);
```
Comment (yonatan-dp) [inline src/components/AboutHome/AboutHome.js]: warningMessage is not being used anywhere else. Why define it as a state? Perhaps use the string directly in displayWarning or only use the local variable defined right before it.

File: src/components/AboutHome/AboutHome.js [hunk 3]
```diff
@@ -49,6 +52,21 @@ const AboutHome = ({ user, setCurrentLink }) => {
     );
   }, []);
 
+  //Display login warnings for the login initial authorizer user with no responsibilities.
+  useEffect(() => {
+    //get the current user
+    const cdxUser = JSON.parse(localStorage.getItem("ecmps_user"));
+
+    //check if the user has initial authorizer role with no responsibilities
+    if (cdxUser?.roles?.includes(config.app.initialAuthorizerRole) &&  JSON.stringify(cdxUser?.facilities) === '{}')
```
Comment (yonatan-dp) [inline src/components/AboutHome/AboutHome.js]: Doing 'JSON.stringify(cdxUser?.facilities)' to just check if facilities is empty may have overhead (among other things).  It also doesn't catch cases where facilities is null or undefined. 

Better alternative:  Object.keys(cdxUser?.facilities || {}).length === 0

For bypass users with no facilities, I think cdxUser.facilities will be null.  Can you check if you are supposed to display the warning message in such cases?

**Review by yonatan-dp — CHANGES_REQUESTED** [review]
Please see comments...

File: src/components/AboutHome/AboutHome.js (hunk 3 above)
Comment (EvanSun220) [reply src/components/AboutHome/AboutHome.js]: Hi Yonatan, the alternative way you provide is a better way to handle facilities empty/null/undefined situation. I will update them based on your notes. 
Also, Mark confirmed that we should be consistent in the behavior, so we will need to display the warning message for bypass users with no facilities.

**Review by yonatan-dp — APPROVED** [review]

**Pull request merged**